### PR TITLE
virtio: console and vsock support

### PR DESCRIPTION
**Pull request by andybui01**
Add new header files for virtio console and vsock work.

### CODE REVIEW

**Comment by andybui01** [comment]
Fails the license check, but the Red Hat licenses are there, untouched.

**Comment by lsf37** [comment]
> Fails the license check, but the Red Hat licenses are there, untouched.

It's not enough for the file to contain any license header, it must contain an `SPDX-Licence-Identifier` declaration and one or  more copyright lines that are machine readable. For the case here, I'd first add a comment

```c
/*
 *  SPDX-License-Identifier: BSD-3-Clause
 */
```

above or below the existing header. If the tool still complains, you might have to include (duplicate) the copyright lines there as well. It probably doesn't like the `(C)` bit and that the year comes at the end, i.e. it'd be

```c
/*
 *  SPDX-License-Identifier: BSD-3-Clause
 * Copyright 2011 Red Hat, Inc.
 * Copyright 2011 Amit Shah <amit.shah@redhat.com> 
 */
```

**Comment by andybui01** [comment]
> It's not enough for the file to contain any license header, it must contain an `SPDX-Licence-Identifier` declaration and one or more copyright lines that are machine readable. For the case here, I'd first add a comment

Noted, thanks!

File: libethdrivers/CMakeLists.txt [hunk 1]
```diff
@@ -117,4 +117,4 @@ string(
     "--undefined=odroidc2_ethernet_ptr"
 )
 
-target_link_libraries(ethdrivers "${DriverModules}")
+target_link_libraries(ethdrivers virtio "${DriverModules}")
```
Comment (kent-mcleod) [inline libethdrivers/CMakeLists.txt:120]: It's probably better to use:
`target_include_directories(ethdrivers PRIVATE $<TARGET_PROPERTY:virtio,INTERFACE_INCLUDE_DIRECTORIES>)`

This more explicitly only uses the library for the header files it exports.

**Review by abrandnewusername — APPROVED** [review]
Looks great to me, please fix the style before merging.

**Review by kent-mcleod — APPROVED** [review]

**Pull request merged**